app/models/krikri/activity.rb

```diff
@@ -27,13 +26,34 @@ def set_end_time
       update_attribute(:end_time, now)
     end
 
+    ##
+    # Runs the block, setting the start and end time of the run. The given block
+    # is passed an instance of the agent, and a URI representing this Activity.
     def run
       if block_given?
         set_start_time
-        yield
+        yield agent_instance, rdf_subject
         set_end_time
       end
     end
 
+    ##
+    # Instantiates and returns an instance of the Agent class with the values in
+    # opts.
+    #
+    # @return [Agent] an instance of the class stored in Agent
+    def agent_instance
+      @angent_instance ||= agent.constantize.new(parsed_opts)
```

Comment:
It's an attribute of the model.  It's defined via ActiveRecord magic in the database table.